lib/search/search_page.dart

```diff
@@ -76,119 +80,16 @@ class SearchPage extends ConsumerWidget {
-                                  ),
-                                ),
-                                Container(
-                                  margin: EdgeInsets.only(top: 16.0),
-                                  child: ElevatedButton(
-                                      style: ElevatedButton.styleFrom(
-                                        backgroundColor: Theme.of(context)
-                                            .colorScheme
-                                            .onSurface,
-                                        foregroundColor: Theme.of(context)
-                                            .colorScheme
-                                            .surface,
-                                        padding: EdgeInsets.symmetric(
-                                            horizontal: 24, vertical: 20),
-                                        shape: RoundedRectangleBorder(
-                                          borderRadius:
-                                              BorderRadius.circular(8.0),
-                                        ),
-                                      ),
-                                      child: Text(
-                                        "Search",
-                                      ),
-                                      onPressed: () async {
-                                        // if (searchCtrl.text.isEmpty) return;
-                                        // var url = Uri.parse(
-                                        //     'https://screen-od6zwjoy2a-an.a.run.app/?name=${searchCtrl.text.toLowerCase()}');
-                                        // var response = await http.post(url, body: {
-                                        //   // 'name': 'doodle',
-                                        //   // 'color': 'blue'
-                                        // });
-                                        // print(
-                                        //     'Response status: ${response.statusCode}');
-                                        // print('Response body: ${response.body}');
-
-                                        // FirebaseFirestore.instance
-                                        //     .collection('search')
-                                        //     .doc(searchCtrl.text)
-                                        //     .set({
-                                        //   'target': searchCtrl.text,
-                                        //   'timeCreated':
-                                        //       FieldValue.serverTimestamp(),
-                                        //   'author': FirebaseAuth
-                                        //       .instance.currentUser!.uid,
-                                        // });
-
-                                        setSearchValue();
-                                      }),
-                                )
+                                buildSearchBar(setSearchValue),
```

Comment:
I don't think this is the right way of doing it. Why do we need to pass a local function? Instead let's have it as a method that can be called in those methods.